File: 4pmclass/DifferentAdapters/app/src/main/java/com/example/differentadapters/adapter/RecyclerAdapter.java
package com.example.differentadapters.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.differentadapters.R;
import com.example.differentadapters.models.ModelItem;

import java.util.ArrayList;

public class RecyclerAdapter  extends RecyclerView.Adapter<RecyclerAdapter.MyViewHOlder> {
ArrayList<ModelItem> userlist;
Context context;

    public RecyclerAdapter(ArrayList<ModelItem> userlist, Context context) {
        this.userlist = userlist;
        this.context = context;
    }

    @Override
    public RecyclerAdapter.MyViewHOlder onCreateViewHolder( ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.recyclerformat,parent,false);
        return new MyViewHOlder(view);
    }

    @Override
    public void onBindViewHolder( RecyclerAdapter.MyViewHOlder holder, int position) {
            ModelItem modelItem = userlist.get(position);
            holder.emailview.setText(modelItem.getUsernumber());
            holder.imageView.setImageResource(modelItem.getImageurl());
    }

    @Override
    public int getItemCount() {
        return userlist.size();
    }

    public class MyViewHOlder extends RecyclerView.ViewHolder {

        ImageView imageView;
        TextView emailview;

        public MyViewHOlder( View itemView) {
            super(itemView);
            imageView = itemView.findViewById(R.id.recyclerimagae);
            emailview = itemView.findViewById(R.id.useremail);
        }
    }
}

File: 4pmclass/MyFirebaseKotlin/app/src/main/java/com/example/myfirebasekotlin/RecyclerAdapter.kt
package com.example.myfirebasekotlin

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.*
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.example.myfirebasekotlin.models.RegisterModel


class RecyclerAdapter(var arraylist:MutableList<RegisterModel>,var context: Context) : RecyclerView.Adapter<RecyclerAdapter.MyviewhOlder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerAdapter.MyviewhOlder {
       var view = LayoutInflater.from(parent.context).inflate(R.layout.layoutrecylerformat,parent,false)
        return MyviewhOlder(view)
    }



    override fun getItemCount(): Int {
        return arraylist.size
    }

    override fun onBindViewHolder(holder: RecyclerAdapter.MyviewhOlder, position: Int) {

        var recyclermod: RegisterModel = arraylist.get(position)
//      holder.imageView.setImageResource(recyclermod.imagerulr)
        holder.numbettext.text = recyclermod.number
        holder.emailtext.text = recyclermod.email

        if(recyclermod.imageurl.equals("default"))
        {
            holder.imageView.setImageResource(R.mipmap.ic_launcher)
        }else
        {
            Glide.with(context).load(recyclermod.imageurl).into(holder.imageView)
        }

    }

    class MyviewhOlder(var itemview: View) : RecyclerView.ViewHolder(itemview)  {

         var imageView:ImageView
      var emailtext:TextView
        var numbettext: TextView

        init {
            imageView = itemview.findViewById(R.id.recyclrimage);
          emailtext = itemview.findViewById(R.id.emailviewrecycler)
            numbettext = itemview.findViewById(R.id.numberviewrecycler)
        }
    }
}